netket/callbacks/earlystopping.py

```diff
@@ -1,25 +1,40 @@
-from numpy import infty
+# Copyright 2020, 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
+from dataclasses import dataclass
+from typing import Union
 
+import numpy as np
+
+
+@dataclass
 class EarlyStopping:
-    """A simple callback to stop NetKet if there are no improvements in the training"""
+    """A simple callback to stop NetKet if there are no more improvements in the training.
+    based on `driver._loss_name`."""
 
-    def __init__(self, min_delta=0, patience=0, baseline=None):
-        """
-        Constructs a new EarlyStopping object that monitors whether a driver is improving
-        over optimisation epochs based on `driver._loss_name`.
+    min_delta: float = 0.0
+    """Minimum change in the monitored quantity to qualify as an improvement."""
+    patience: Union[int, float] = 0
+    """Number of epochs with no improvement after which training will be stopped."""
```

Comment:
Is this a sane default? wouldn't `np.infty` be a better one?

Reply:
I know this is the one that @VolodyaCO had chosen previously.

Reply:
With `patience = infty`, the callback will wait for infinitely many steps of non-improvement, essentially disabling this check (which is okay if one only wants the baseline to matter - as I've done for the variance criterion in the example - but otherwise that's not a sensible default).